pub/lib/dependabot/pub/metadata_finder.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+require "excon"
+require "dependabot/metadata_finders"
+require "dependabot/metadata_finders/base"
+require "dependabot/shared_helpers"
+
+module Dependabot
+  module Pub
+    class MetadataFinder < Dependabot::MetadataFinders::Base
+      private
+
+      def look_up_source
+        repo = pub_listing.dig("latest", "pubspec", "repository")
+        return nil unless repo
+
+        Source.from_url(repo)
+      end
```

Comment:
I think this will actually give us a bunch of Dependabot's default behavior out of the box, so this is 👍